raft/handler_test.go

```diff
@@ -1,67 +1,79 @@
 package raft_test
 
 import (
-	"encoding/binary"
+	"io"
+	"io/ioutil"
 	"net/http"
+	"net/http/httptest"
 	"net/url"
-	"reflect"
 	"testing"
-	"time"
 
 	"github.com/influxdb/influxdb/raft"
 )
 
 // Ensure a node can join a cluster over HTTP.
-func TestHTTPHandler_HandleJoin(t *testing.T) {
-	n := NewInitNode()
-	defer n.Close()
+func TestHandler_HandleJoin(t *testing.T) {
+	h := NewHandler()
+	h.AddPeerFunc = func(u *url.URL) (uint64, *raft.Config, error) {
+		if u.String() != "http://localhost:1000" {
+			t.Fatalf("unexpected url: %s", u)
+		}
+		return 2, &raft.Config{}, nil
+	}
+	s := httptest.NewServer(h)
```

Comment:
Does this server involve actual binding to port 1000? Because that is not allowed without `root` privileges.